### ISSUE TITLE
When tunneling sshd service, we have double encapsulation of SSH.

### ISSUE THREAD

**Issue by BoleynSu** [jpillora/chisel#188]
Please refer to https://github.com/jpillora/chisel/pull/187 for more details.

Note that, if we implemented the feature to avoid double encapsulation we still need to keep the built-in sshd service for encryption to avoid leaking unencrypted data. That is we need to explicitly enable the feature instead of making it the default.

**Comment by jpillora** [comment]
RE: Code from https://github.com/jpillora/chisel/pull/187#issuecomment-687523044

I originally was doing this, and my initial PoCs looked very similar to this, however I wanted to multiplex TCP connections to mask the true number of connections going over the wire. So initially I added https://github.com/hashicorp/yamux, and that worked fairly well, then I realised I needed encryption, so I rolled my own thing https://github.com/jpillora/conncrypt which was a terrible idea, then I realised that SSH had both connection multiplexing and really nice crypto, and decided to accept the performance hit from the SSH protocol.

So basically, since I want TCP multiplexing in chisel - you need the extra layer of encapsulation ontop of websockets. Next, if we want to remove the double encryption, we'd need to either (1) have an alternative TCP multiplexing library (go back to yamux), and add the complexity of choosing between that and SSH (2) use a NULL cipher in SSH, and last time I checked - this currently isn't supported, which means we'd need to maintain a fork.

All of the above is a fairly large amount of work to improve the performance of the double encryption case. And it'd be nice to have a set of benchmarks to prove it's worth the effort
